### PR TITLE
O11Y integration

### PR DESCRIPTION
**Pull request by paulomach**
## Issue

COS support on k8s.
[DPE-1558](https://warthogs.atlassian.net/browse/DPE-1558)

## Solution

Integrated mysql_exporter from snap/rock and usage of o11y libraries to expose metrics and log files

[DPE-1558]: https://warthogs.atlassian.net/browse/DPE-1558?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: metadata.yaml [hunk 1]
```diff
@@ -41,12 +41,21 @@ provides:
...
 requires:
   certificates:
     interface: tls-certificates
     limit: 1
+    optional: true
   s3-parameters:
     interface: s3
     limit: 1
+    optional: true
+  logging:
```
Comment (shayancanonical) [inline metadata.yaml:57]: can this interface use a `limit`? same with the `grafana-dashboard` and `metrics-endpoint`?

File: src/charm.py [hunk 2]
```diff
@@ -95,6 +103,18 @@ def __init__(self, *args):
...
         self.backups = MySQLBackups(self, self.s3_integrator)
+        self.grafana_dashboards = GrafanaDashboardProvider(self)
+        self.metrics_endpoint = MetricsEndpointProvider(
+            self,
+            refresh_event=self.on.start,
+            jobs=[{"static_configs": [{"targets": [f"*:{MYSQLD_EXPORTER_PORT}"]}]}],
+        )
+        self.loki_push = LogProxyConsumer(
+            self,
+            log_files=["/var/log/mysql/error.log"],
```
Comment (shayancanonical) [inline src/charm.py:114]: use constant for error log file

File: metadata.yaml (hunk 1 above)
Comment (paulomach) [reply metadata.yaml:57]: 2148b51

File: src/charm.py (hunk 2 above)
Comment (paulomach) [reply src/charm.py]: 33229df

File: metadata.yaml [hunk 3]
```diff
@@ -41,12 +41,22 @@ provides:
     interface: mysql_client
   osm-mysql:
     interface: mysql
+  metrics-endpoint:
+    interface: prometheus_scrape
+  grafana-dashboard:
+    interface: grafana_dashboard
```
Comment (shayancanonical) [inline metadata.yaml:47]: just confirming that we dont want to limit these relations as well?

**Review by shayancanonical — APPROVED** [review]
LGTM, but I had a one minor confirmation comment

**Review by delgod — APPROVED** [review]

File: metadata.yaml (hunk 3 above)
Comment (paulomach) [reply metadata.yaml:47]: I'm not sure yet - I'll leave for another PR

**Pull request merged**